### PR TITLE
[#36] [Chore] Add UI integration testing - (3/3) Taking Survey test

### LINKED ISSUE #36: [Chore] Add UI integration testing - (3/3) Taking Survey test
## Why
Add UI integration testing to make sure the app's flow works as expected.

## Acceptance Criteria
- Add a flow test case with the following steps:
  - Navigate to the Survey Detail screen after selecting a survey item on the Home screen. (#35)
  - Tap the **"Start Survey"** button.
  - Select the answers or fill in the information in the forms.
  - Tap the **"Submit"** button.
  - Show the Complete Survey screen, then navigate to the Home screen.

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
- Closes #36 

## What happened 👀

Add UI integration testing to make sure the app's flow works as expected.

## Insight 📝

- Add tests for choice, textArea, textField, nps, smiley, star, thumbs, heart, dropdown, intro, outro, submit success and failure events

## Proof Of Work 📹

https://github.com/nimblehq/flutter-ic-kaung-thieu/assets/32578035/22477a8a-9192-48e3-967d-1939cdeaf90f

### CODE REVIEW

File: integration_test/survey_detail_screen_test.dart [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+      await tester.pumpAndSettle();
+
+      final startSurveyButton = find.byType(ElevatedButton);
+      await tester.tap(startSurveyButton);
+      await tester.pumpAndSettle();
+
+      expect(find.byType(CupertinoPicker), findsOneWidget);
+    });
+
+    testWidgets('When survey reached the end, it shows outro type question',
```
Comment (manh-t) [inline integration_test/survey_detail_screen_test.dart:535]:
```suggestion
    testWidgets('When survey questions reached the end, it shows outro type question',
```

**Review by manh-t — COMMENTED** [review]
rest LGTM

File: integration_test/survey_detail_screen_test.dart [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+      await tester
+          .pumpWidget(TestUtil.pumpWidgetWithShellApp(const SurveyDetailScreen(
+        surveyId: 'id',
+      )));
+      await tester.pumpAndSettle();
+
+      expect(
+        find.byType(ElevatedButton),
+        findsOneWidget,
+      );
```
[the remark this answers is not in the document]
Comment (kaungkhantsoe) [reply integration_test/survey_detail_screen_test.dart:48]: @manh-t Added in https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/106/commits/6013e7ccd090c91b001c04508f13ce72291ba68c

**Review by ducbm051291 — APPROVED** [review]
LGTM 👍

**Review by manh-t — APPROVED** [review]

**Pull request merged**